spec/controllers/alaveteli_pro/subscriptions_controller_spec.rb

```diff
@@ -189,6 +189,43 @@
         end
       end
 
+      context 'with an existing customer and an incomplete subscription' do
+
+        let(:customer) do
+          Stripe::Customer.create(
+            email: user.email,
+            source: stripe_helper.generate_card_token
+          )
+        end
+
+        let(:pro_account) do
+          user.create_pro_account(stripe_customer_id: customer.id)
+        end
+
+        it 'should cancel any incomplete subscriptions' do
+          # we can't create a subscription in the incomplete status so we have
```

Comment:
Could we do something similar to https://github.com/mysociety/alaveteli/blob/develop/lib/stripe_mock_patch.rb#L6-L10 ?

Reply:
Possibly yes. Want me to give that a go?

Reply:
Worth having a stab at for the remaining hour of the day. If we can't get it working by then, I think we just go with this.